tests/xdc/advanced_visibility_test.go

```diff
@@ -317,8 +351,26 @@ func (s *esCrossDCTestSuite) TestSearchAttributes() {
 		s.True(verified)
 	}
 
+	saListRequest = &workflowservice.ListWorkflowExecutionsRequest{
+		Namespace: namespace,
+		PageSize:  int32(2),
+		Query:     fmt.Sprintf(`WorkflowId = "%s" and %s = "another string"`, id, s.testSearchAttributeKey),
```

Comment:
do we have a programmatic query builder yet, or can we factor one out of the new visibility? code like this is kind of scary, even though it's in tests

Reply:
What's wrong with it? It is functional test. It supposed to emulate user input.

Reply:
this is fine, but there are a few instances in non-test code now

Reply:
That's on me. I'll work on making those parts better.